### PR TITLE
Repository Config list page

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Updates the Repository Configuration list page to implement the latest designs. There's still some work to do, but wanted to get this as a first milestone to keep the PR from growing too big. Most of the look and feel are here, and basic functionality of searching/paging should work as well. There are some details around loading indicators, and pagination related state that I plan on addressing in the next round of this UI.

| Header | Header | Header |
|--------|--------|--------|
| image | image | image | 

Things that aren't accounted for yet but will be in followups. I've called them out in comments in the code, but for summary:

* Need the prebuild status filter
* Ability to change rows per page (it's set to 5 now to ease development).
* Search input is missing some details (icon, proper search input type). I plan on pulling this into podkit and formalizing it a bit from our current TextInput component.
* Import modal needs attention and swapping to use new grpc api
* Pagination/Search state handling (need to update api response to include some more pagination details and need to shift state into query params instead of component state).

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 09d9d9e</samp>

Fixed pagination bug in `getProjects` method of `ConfigurationServiceAPI`. This method lists the user's projects in the dashboard.

</details>

## Related Issue(s)

Fixes EXP-876

## How to test

* Import at least 6 repositories (so you have some paging)
* Play with paging a bit (there are some edge cases I need to address still but happy path should work)
* Make sure dark mode & light mode look right w/ the Table components

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-exp-84ec419e928</li>
	<li><b>🔗 URL</b> - <a href="https://brad-exp-84ec419e928.preview.gitpod-dev.com/workspaces" target="_blank">brad-exp-84ec419e928.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - brad-exp-876-repositories-list-page-gha.19667</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-brad-exp-84ec419e928%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Review by akosyakov — APPROVED** [review]
I look at service side look good to me, i did not look at frontend code or tested it. cc @filiptronicek could you do it please?

File: components/dashboard/src/repositories/list/PaginationControls.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+type Props = {
+    currentPage: number;
+    totalPages: number;
+    pageSize: number;
+    totalRows: number;
+    currentRows: number;
+    onPageChanged: (page: number) => void;
+};
+export const PaginationControls: FC<Props> = ({
```
[the remark this answers is not in the document]
Comment (selfcontained) [reply components/dashboard/src/repositories/list/PaginationControls.tsx:21]: Thanks! That was my thought too. I'll wait until more of it's flushed out and then we can lift it up.

File: components/gitpod-db/src/project-db.ts [hunk 2]
```diff
@@ -37,3 +31,12 @@ export interface ProjectDB extends TransactionalDB<ProjectDB> {
     getProjectUsage(projectId: string): Promise<ProjectUsage | undefined>;
     updateProjectUsage(projectId: string, usage: Partial<ProjectUsage>): Promise<void>;
 }
+
```
Comment (selfcontained) [inline components/gitpod-db/src/project-db.ts:34]: lifted these args into an object as the ordering/number of them was getting pretty hairy, especially w/ a couple being optional.

File: components/server/src/projects/projects-service.ts [hunk 3]
```diff
@@ -70,15 +70,24 @@ export class ProjectsService {
...
+
+        const projects = await this.projectDB.findProjectsBySearchTerm({
+            offset: searchOptions.offset || 0,
+            limit: searchOptions.limit || 1000,
+            orderBy: searchOptions.orderBy || "creationTime",
+            orderDir: searchOptions.orderDir || "ASC",
+            searchTerm: searchOptions.searchTerm || "",
+            organizationId: searchOptions.organizationId,
+        });
+        // TODO: adjust this to not filter entities, but log errors if any are not accessible for current user
```
Comment (selfcontained) [inline components/server/src/projects/projects-service.ts:90]: I'm going to tackle this in a followup and leave it filtering for now

File: components/server/src/api/configuration-service-api.ts [hunk 4]
```diff
@@ -78,9 +78,11 @@ export class ConfigurationServiceAPI implements ServiceImpl<typeof Configuration
         }
 
         const limit = req.pagination?.pageSize || 25;
```
Comment (akosyakov) [inline components/server/src/api/configuration-service-api.ts:80]: Should we have max page size and return invalid argument if a user requests more?

**Review by filiptronicek — APPROVED** [review]
Found this to work reliably — nice work! I haven't tested multiple users per org, but because all the repos are currently public, I don't believe that will make a difference. ([org invite](https://brad-exp-84ec419e928.preview.gitpod-dev.com/orgs/join?inviteId=b529e245-9d45-433e-9938-4c389ecc006f) in case you'd want to last-minute test it yourself :).

Changing orgs behaves as expected and the pagionation as well.

**Comment by filiptronicek** [comment]
After restarting my browser, after changing orgs again, I did not initially get any. Could have been a bug though.

File: components/server/src/api/configuration-service-api.ts (hunk 4 above)
Comment (selfcontained) [reply components/server/src/api/configuration-service-api.ts:80]: For sure, I'll add this in a followup, need to do some work around pagination/cursors anyways.

Comment (akosyakov) [reply components/server/src/api/configuration-service-api.ts:80]: yeah, sorry for the confusion in this area yet, i hope we can settle it today

Comment (akosyakov) [reply components/server/src/api/configuration-service-api.ts:80]: I don't think it is blocking since it is under FF, we can resolve with follow up PR?
